lib/solidus_stripe/webhook/event.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module SolidusStripe
+  module Webhook
+    # Omnes event wrapping a Stripe event.
+    #
+    # All unknown methods are delegated to the wrapped Stripe event.
+    #
+    # @see Stripe::Event
+    class Event
+      include Omnes::Event
+
+      # TBD
+      CORE_EVENTS = Set[*%i[]].freeze
+      private_constant :CORE_EVENTS
+
+      # @api private
+      def self.register(user_events:, bus:, core_events: CORE_EVENTS)
+        (core_events + user_events).each do |event|
+          bus.register(:"stripe.#{event}")
+        end
```

Comment:
If everything is going to be handed to the user application, then we can remove the distinction. However, if we provide default implementations, that's not the case, right? 🤔 Maybe I'm missing something, but we probably need to capture incoming webhooks for payment capture so that our models are updated accordingly. That's not something I imagine belonging to the copied templates but to the core models. Am I wrong?

Reply:
If possible I would err on the side of allowing the target app to take control of core events. Stripe has many features and maybe an additional fraud check or external system integration needs to be added.

The implementations we provide out of the box should be more examples and (I think) allow breaking them apart leaving the glue code in the target application. That to me would be the best of both worlds.

I'm assuming it's possible to do that, but if not maybe the current division is already the best we can do

Reply:
As you can create more than one subscriber for a given event, we're good at letting users add new stuff.

As per our subscribers, we should design them with the SRP in mind. For instance, if we need to do two things on `payment.succeeded`, let's create two subscribers for the same event and not a single one. This way, we pinpoint the need for any of the regular Solidus approaches to modify core behavior. On top of that, for subscribers, Omnes allows you to [unsubscribe](https://github.com/nebulab/omnes#unsubscribing--clearing) listeners from the bus 👍